bundles/config/org.eclipse.smarthome.config.dispatch/src/main/java/org/eclipse/smarthome/config/dispatch/internal/ConfigDispatcher.java

```diff
@@ -260,8 +213,8 @@ private void readDefaultConfig() {
         }
     }
 
-    private void readConfigFiles() {
-        File dir = getSourcePath().toFile();
+    private void readConfigFiles(Path pathToWatch) {

```

Comment:
In the ConfigDispatcher the path should be named like `basePath` since the CD does not watch paths any more.

Reply:
On a second thought the initial processing of the files in the base config dir might also be a good thing the new `ConfigDispatcherFileWatcher` could do after activation. It would simply call `configDispatcher.processFile` in the for loop. Or to make the separation even more fine granular: Let the CDFileWatcher list the files in the base config dir and pass it to a new `public void processFiles(List<File> files)` on the ConfigDispatcher which does the sorting and processing.
wdyt?

Reply:
Since the ConfigDispatcher has a strict ordering in its constructor the calls to `processOrphanExclusivePIDs()` and `storeCurrentExclusivePIDList()` should be done afterwards in the new `processFiles(List<File> files)` method.
This will also slim down the c'tor of the ConfigDispatcher :-)